docs/spec.rst

```diff
@@ -0,0 +1,501 @@
+JSpecify 0.1 spec outline
+=========================
+
+About this document
+-------------------
+
+This document summarizes our current thinking for proposing a set of
+nullness annotations. **Nothing here is “set in stone” yet.**
+
+It will be very helpful to read or refer to the `core
+concepts/glossary <https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit>`__
+document in order to understand this material correctly. There are
+numerous other documents in our `shared
+folder <https://drive.google.com/drive/folders/1vZl1odNCBncVaN7EwlwfqI05T_CHIqN->`__.
+
+Overview
+--------
+
+We propose a set of type annotations to specify the intended nullness of
+individual type usages. At a high level, our proposed semantics is
+similar to existing tools’ treatment of nullness type annotations. We
+treat nullness annotations as purely compile-time type refinements
+without effect on runtime semantics. Besides the typical “nullable” and
+“non-null” refinements, we distinguish values of “unspecified nullness”,
+which typically arise from unannotated code, and allow for (with
+optional warnings if desired) unsound subtyping between “unspecified
+nullness” and known nullness states (i.e., nullable or non-null).
+
+To avoid excessive annotation overhead in hand-written Java code while
+clearly distinguishing annotated from legacy code, we also propose a set
+of explicit defaulting annotations. In the absence of a defaulting
+annotation, type usages will be considered of “unspecified nullness.”
+
+We give `semantics <#semantics>`__ to these annotations independent of
+particular tools. We don’t specify what tools must do with the
+information we provide, nor do we forbid them from adding to it (e.g.,
```

Comment:
"we provide" -> "provided using annotations defined here" or something like that?

Reply:
Tweaked slightly differently to try to keep the length down. But maybe it's less clear now? I can give it another shot if you think I should.